libs/stokesian_dynamics/multi_array.hpp

```diff
@@ -0,0 +1,248 @@
+#pragma once
+
+#include <cmath>
+#include <stdexcept>
+#include <string>
+#include <type_traits>
+
+#ifdef __CUDACC__
```

Comment:
Does Clang define this when compiling Cuda code?

Reply:
According to the [Clang docs](https://releases.llvm.org/9.0.0/docs/CompileCudaWithLLVM.html#detecting-clang-vs-nvcc-from-code), yes:
> When clang is actually compiling CUDA code - rather than being
> used as a subtool of NVCC’s - it defines the \_\_CUDA\_\_ macro.

This ifdef should probably be `#if defined(__HIPCC__) or defined(__CUDACC__)` to avoid issues with ROCm (as per [src/utils/include/utils/device_qualifier.hpp](https://github.com/espressomd/espresso/blob/python/src/utils/include/utils/device_qualifier.hpp))